### PR TITLE
Refactor client auth service

### PR DESCRIPTION
**Pull request by Awk34**
When using `Util.safeCb`, we also checked if `callback` was not a function, and used `angular.noop` if it wasn't. This change will instead error if `callback` is not a function, which I believe will avoid some confusion.

### CODE REVIEW

File: app/templates/client/components/auth(auth)/auth.service.js [hunk 1]
```diff
@@ -162,8 +161,7 @@ function AuthService($location, $http, $cookies, $q, appConfig, Util, User) {
       * Check if a user is an admin
       *   (synchronous|asynchronous)
       *
-      * @param  {Function|*} callback - optional, function(is)
```
Comment (kingcody) [inline app/templates/client/components/auth(auth)/auth.service.js]: This param gets passed to `hasRole` so the signature is valid.

File: app/templates/client/components/auth(auth)/auth.service.js [hunk 2]
```diff
@@ -20,7 +19,7 @@ function AuthService($location, $http, $cookies, $q, appConfig, Util, User) {
      * @param  {Function} callback - optional, function(error, user)
      * @return {Promise}
      */
-    login: function(user, callback) {
+    login: function(user, callback = angular.noop) {
```
Comment (kingcody) [inline app/templates/client/components/auth(auth)/auth.service.js]: This will cause line 33 to fail if `callback && typeof callback !== 'function'`.

Comment (Awk34) [inline app/templates/client/components/auth(auth)/auth.service.js]: Yeah, this is on purpose. If a use puts in `'oogabooga'` for `callback`, we want an error, not to silently eat it.

File: app/templates/client/components/auth(auth)/auth.service.js (hunk 1 above)
Comment (Awk34) [inline app/templates/client/components/auth(auth)/auth.service.js]: fixed

File: app/templates/client/components/auth(auth)/auth.service.js (hunk 2 above)
Comment (kingcody) [inline app/templates/client/components/auth(auth)/auth.service.js]: By design if the users passes in anything then they are opting for async behavior and we return a promise.

Comment (Awk34) [inline app/templates/client/components/auth(auth)/auth.service.js]: It doesn't seem reasonable to me for anything but a function to be supplied to the function, though

Comment (kingcody) [inline app/templates/client/components/auth(auth)/auth.service.js]: So to use the promise behavior a user would have to run `Auth.getCurrentUser(function() {}).then();` ? Why not just `Auth.getCurrentUser(true).then();` ?

Comment (Awk34) [inline app/templates/client/components/auth(auth)/auth.service.js]: I'm confused. Won't we call the callback **and** return a promise either way?

File: app/templates/client/components/auth(auth)/auth.service.js [hunk 3]
```diff
@@ -95,7 +94,7 @@ function AuthService($location, $http, $cookies, $q, appConfig, Util, User) {
      * @param  {Function|*} callback - optional, funciton(user)
      * @return {Object|Promise}
      */
-    getCurrentUser: function(callback) {
+    getCurrentUser: function(callback = angular.noop) {
       if (arguments.length === 0) {
```
Comment (kingcody) [inline app/templates/client/components/auth(auth)/auth.service.js]: This method is only async when `arguments.length === 0`

File: app/templates/client/components/auth(auth)/auth.service.js [hunk 4]
```diff
@@ -104,10 +103,10 @@ function AuthService($location, $http, $cookies, $q, appConfig, Util, User) {
         currentUser.$promise : currentUser;
       return $q.when(value)
         .then(function(user) {
-          safeCb(callback)(user);
+          callback(user);
```
Comment (kingcody) [inline app/templates/client/components/auth(auth)/auth.service.js]: If the user calls `Auth.getCurrentUser(null).then()` this line will fail.

**Pull request closed without merging**